### PR TITLE
Unit tests tumortypes txt

### PR DESCRIPTION
**Pull request by sheridancbio**
Add unit test for TumorTypes_txt util
Also add MainType tests for TumorTypes util

### CODE REVIEW

**Comment by sheridancbio** [comment]
This PR is not yet complete .. and in any case should not be merged before #60 (which is already incorporated into this branch).

To complete this PR, the children attributes will be set in the construction of the expectedTumorTypes from the TestConfig, so that they can be tested directly in the TumorTypes test, and also so that they can be used during the construction of the InputStream during testing of the tumorTypes_txt util.

**Comment by sheridancbio** [comment]
This PR now can be merged once #60 has been approved/merged. The unit test can be improved -- for now it is testing that the output TSV spreadsheet has the correct number of columns, contains the correct header, and has the correct number of data lines based on the (mocked) oncoTree full repository.

File: core/src/test/java/org/mskcc/oncotree/utils/OncotreeUtilTestConfig.java [hunk 1]
```diff
@@ -37,48 +36,48 @@
...
+
     @Bean
     public ArrayList<OncoTreeNode> oncoTreeRepositoryMockResponse() throws Exception {
         return setupOncotreeRepositoryMockResponse();
     }
-    
+
     @Bean
-    public Map<String, TumorType> expectedTumorTypeMap() throws Exception {
+    public HashMap<String, TumorType> expectedTumorTypeMap() throws Exception {
```
Comment (zheins) [inline core/src/test/java/org/mskcc/oncotree/utils/OncotreeUtilTestConfig.java]: Why is this necessary?

File: core/src/test/java/org/mskcc/oncotree/utils/OncotreeUtilTestConfig.java [hunk 2]
```diff
@@ -37,48 +36,48 @@
...
         return setupExpectedTumorTypeMap();
     }
-    
+
     @Bean
-    public List<MainType> expectedMainTypeList() throws Exception {
+    public ArrayList<MainType> expectedMainTypeList() throws Exception {
```
Comment (zheins) [inline core/src/test/java/org/mskcc/oncotree/utils/OncotreeUtilTestConfig.java]: Why is this necessary?

File: core/src/test/java/org/mskcc/oncotree/utils/OncotreeUtilTestConfig.java [hunk 3]
```diff
@@ -118,7 +116,7 @@ private Version setupMockVersion() {
         if (caseCount < 1) {
             throw new Exception("Error : no test cases defined in rawTestValueSource");
         }
-        expectedTumorTypeMap = new HashMap<>(caseCount + 1);
+        HashMap<String, TumorType> expectedTumorTypeMap = new HashMap<>(caseCount + 1);
```
Comment (zheins) [inline core/src/test/java/org/mskcc/oncotree/utils/OncotreeUtilTestConfig.java]: Why is this necessary?

File: core/src/test/java/org/mskcc/oncotree/utils/OncotreeUtilTestConfig.java [hunk 4]
```diff
@@ -147,13 +145,12 @@ private Version setupMockVersion() {
...
             expectedTumorTypeMap.put(code, nextType);
         }
-        return expectedTumorTypeMap;        
+        return expectedTumorTypeMap;
     }
-    
-    
-    private List<MainType> setupExpectedMainTypeList() throws Exception {
+
+    private ArrayList<MainType> setupExpectedMainTypeList() throws Exception {
```
Comment (zheins) [inline core/src/test/java/org/mskcc/oncotree/utils/OncotreeUtilTestConfig.java]: Why is this necessary?

File: core/src/test/java/org/mskcc/oncotree/utils/OncotreeUtilTestConfig.java [hunk 5]
```diff
@@ -147,13 +145,12 @@ private Version setupMockVersion() {
...
         Set<MainType> expectedMainTypeSet = new HashSet<>();
-        List<MainType> expectedMainTypeList = new ArrayList<>();
+        ArrayList<MainType> expectedMainTypeList = new ArrayList<>();
```
Comment (zheins) [inline core/src/test/java/org/mskcc/oncotree/utils/OncotreeUtilTestConfig.java]: Why is this necessary?

File: core/src/test/java/org/mskcc/oncotree/utils/TumorTypesUtilTest.java [hunk 6]
```diff
@@ -45,24 +47,31 @@
...
-    public Set<TumorType> simulateTumorTypeApiResponse() throws Exception {
+    public Set<TumorType> simulateFullTumorTypeApiResponse() throws Exception {
         Map<String, TumorType> tumorTypeMap = tumorTypesUtil.getTumorTypesByVersionFromRaw(mockVersion);
         Set<TumorType> tumorTypeSet = tumorTypesUtil.flattenTumorTypes(tumorTypeMap, null);
         return tumorTypeSet;
     }
 
+    private Map<String, TumorType> getReturnedFullTumorTypeMap() throws Exception {
+        Set<TumorType> tumorTypesAsPreparedForTumorTypesEndpoint = simulateFullTumorTypeApiResponse();
+        HashMap<String, TumorType> returnedTumorTypeMap = new HashMap<>(tumorTypesAsPreparedForTumorTypesEndpoint.size());
```
Comment (zheins) [inline core/src/test/java/org/mskcc/oncotree/utils/TumorTypesUtilTest.java]: Why does this need to be initialized specifically as `HashMap` and not `Map`?

File: core/src/test/java/org/mskcc/oncotree/utils/TumorTypesUtilTest.java [hunk 7]
```diff
@@ -72,22 +81,20 @@ private boolean testValuesMatch(String value1, String value2) {
...
     @Test
     public void testFullOncoTreeCodeListByCode() throws Exception {
-        Set<TumorType> tumorTypesAsPreparedForTumorTypesEndpoint = simulateTumorTypeApiResponse();
-        HashMap<String,TumorType> returnedTumorTypeMap = new HashMap<>(tumorTypesAsPreparedForTumorTypesEndpoint.size());
-        for (TumorType tumorType : tumorTypesAsPreparedForTumorTypesEndpoint) {
-            returnedTumorTypeMap.put(tumorType.getCode(), tumorType);
-        }
         int failureCount = 0;
         StringBuilder failureReport  = new StringBuilder();
+        Map<String,TumorType> returnedTumorTypeMap = getReturnedFullTumorTypeMap();
```
Comment (zheins) [inline core/src/test/java/org/mskcc/oncotree/utils/TumorTypesUtilTest.java]: This one is not a `HashMap` specifically. Is this because it is not what is returned?

File: core/src/test/java/org/mskcc/oncotree/utils/TumorTypesUtilTest.java [hunk 8]
```diff
@@ -102,6 +109,29 @@ public void testFullOncoTreeCodeListByCode() throws Exception {
                     failureReport.append(makeMismatchMessage(oncoTreeCode, "name", returnedName, expectedName));
                     failureCount = failureCount + 1;
                 }
+                MainType expectedMainType = expectedTumorType.getMainType();
+                MainType returnedMainType = returnedTumorType.getMainType();
+                if (expectedMainType == null || returnedMainType == null) {
```
Comment (zheins) [inline core/src/test/java/org/mskcc/oncotree/utils/TumorTypesUtilTest.java]: Do you care if they are both null? I guess do you need this outer if statement here? That last else statement could just go after these two checks if one is null and not the other. Not sure if it matters though, performance could be better this way if we don't expect it to happen often (test passes).

Comment (sheridancbio) [inline core/src/test/java/org/mskcc/oncotree/utils/TumorTypesUtilTest.java]: If either is null, then a null pointer exception could thrown on line 131 when the .getName() method is called
